jscomp/test/js_re_test.ml

```diff
@@ -1,12 +1,31 @@
 
 let suites = Mt.[
+  "matches", (fun _ ->
+    let re = [%re "/(\\d+)-(?:(\\d+))?/g"] in
+    let str = "3-" in
+
+    match re |> Js.Re.exec str with
+      | Some result ->
+        let matches = (Js.Re.matches result) in
+        let someMatch = matches.(2) in
+        let hit = ref false in
+        let _ = Js.Null_undefined.iter someMatch ((fun _ -> hit := true) [@bs]) in
+        Eq(false, !hit)
```

Comment:
I think perhaps it would be simpler and clearer to just compare directly against `Js.null` or `Js.undefined`